examples/drivers/windows/ottmp/adapter.cpp

```diff
@@ -0,0 +1,551 @@
+/*
+ *    Copyright (c) 2016, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+ *    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This module implements code to manage the NETADAPTER object for the
+ *   network adapter.
+ */
+
+#include "pch.hpp"
+#include "adapter.tmh"
+
+PAGED
+_IRQL_requires_max_(PASSIVE_LEVEL)
+NDIS_STATUS
+AdapterInitialize(
+    _In_ NDIS_HANDLE                MiniportAdapterHandle,
+    _In_ POTTMP_ADAPTER_CONTEXT     AdapterContext
+    )
+/*++ 
```

Comment:
I agree it is very different from the rest of the style of OpenThread, but none of the styling in the Windows driver code matches OpenThread. This code is following the Windows driver standard way. See from the [Windows driver samples](https://github.com/Microsoft/Windows-driver-samples/blob/master/network/ndis/netvmini/6x/adapter.c#L140). If this really would be a blocker for merging, I can change it, though I'd prefer not to. Also, this style is already being used for the filter driver code checked in, as you can see from [here](https://github.com/openthread/openthread/blob/master/examples/drivers/windows/otLwf/driver.c#L52) and [here](https://github.com/openthread/openthread/blob/master/examples/drivers/windows/otLwf/filter.c#L56).